### ISSUE TITLE
Nested arguments

### ISSUE THREAD

**Issue by bssergy** [mickhansen/graphql-sequelize#677]
I.e. my schema looks like:
```
products
  user
```
I want to query products whose user has property active: true.
But query like:
```
{
  products(where: { user: { active: true } }) {
    user {
      name
    }
  }
}
```
throws an error: "Unknown value { active: true }"
Does sequelize have support for nested arguments? Are there any workarounds?

**Comment by mickhansen** [comment]
It's not something graphql-sequelize can handle.
You can write your own logic to pass down arguments via the resolver.

**Comment by bssergy** [comment]
@mickhansen ok, I use defaultArgs and resolver helpers. Where clause can be really difficult with operators:
where: { or: [ { user: { active: true } }, productName: "Others" ] }
It's really hard to parse this query manually. Also, there should be included user assosiation. Is it possible to do this automatically?

**Comment by mickhansen** [comment]
I wouldn't suggest using `where` in the graphql schema at all, use descriptive arguments to express your logic instead.

User associations can be handled via the resolver and dataloader, no need to prefetch.

**Comment by bssergy** [comment]
Ok. Just to clarify. If I handle user assosiation in resolver with dataloader, then I will have query like:
```
{
  products: {
    user(active: true) {
      name
    }
  }
}
```
But this query returns me ALL products, and then attach users with property active=true. But I need query ONLY these products, whose user has property active=true. Is it possible to do with graphql-sequelize?

**Comment by mickhansen** [comment]
`products(fromActiveUsers: true) {}` , for the products resolver you add a before hook that looks for that argument and applies logic.

**Comment by janmeier** [comment]
Everything is possible, but you will probably have to write the logic yourself.

The query is doing what it's supposed to. A filter on the child field (user) should _never_ affect the parent field (products). If you want to filter products, you need to add the filters directly to product like above ^^
